### ISSUE TITLE
startup hangs for a looong time when dns servers cannot resolve the current hostname (not the target names)

### ISSUE THREAD

**Issue by v4hn** [duncs/clusterssh#158]
I was wondering why my sessions sometimes take a veeeery long time to start.
After some digging it turns out that there is a macro supported that fills in the current hostname and [looks it up via `Net::Domain::hostfqdn()`](https://github.com/duncs/clusterssh/blob/b302a7724f2ba8bb1c1b5b812b53f5c3abf03198/lib/App/ClusterSSH/Window/Tk.pm#L604) (whether or not it's needed).

However, there are documented issues with this lookup in certain environments because it attempts to look up domain information through DNS request "attempts" instead of resolving the request locally and DNS servers might not reply at all with high timeouts. In my current work-environment this is what happens:
```
$ time perl -MNet::Domain -e 'print Net::Domain::hostfqdn(),"\n"'
this_machines_name

real    1m21.700s
user    0m0.021s
sys     0m0.015s
```

There is a [Munin bug reported almost 20 years ago](https://linux.debian.bugs.dist.narkive.com/JIHomnkx/bug-307462-munin-node-installation-hangs-possible-problem-with-cpu-plugin) that describes the issue in detail. (I never thought I would refer to a bug documentation that old with current relevance, guess I grew old..)

My current workaround is to set `macros_enabled=no` in `.clusterssh/config` and thus avoid the logic.

I suggest to replace the `hostname` macro by `Sys::Hostname::hostname()`, as it seems unexpected that `hostname` should imply the fqdn anyway.

A less invasive alternative might be to at least check whether the macro is required before substitution and maybe add some debug output for it.

**Comment by duncs** [comment]
This should be fixed in 4.18 with the `hostname_override` setting within the configuation file; set it to either the FQDN or IP address of your server and it will avoid doing the DNS lookup.

I also switched the DNS lookup method in line with your suggestion

**Comment by v4hn** [comment]
Thank you for addressing this! :bouquet:
